contracts/multisig-prover/src/state.rs

```diff
@@ -36,19 +37,20 @@ pub const REPLY_BATCH: Item<BatchID> = Item::new("reply_tracker");
 pub struct WorkerSet {
     pub signers: BTreeSet<Signer>,
     pub threshold: Uint256,
-    pub nonce: u64, // for randomness
+    // for hash uniqueness. The same exact worker set could be in use at two different times,
+    // and we need to be able to distinguish between the two
+    pub nonce: u64,
```

Comment:
I prefer to change to `created_at` if we use current block height in `new`, or have a counter to track the nonce.